### DISCUSSION TITLE
How to controll spinner?

### DISCUSSION THREAD

**Discussion by tadejgolobic** [MetaMask/snaps#1899] [Q&A] [answered]
I see that we have following UI component available to use - spinner (https://docs.metamask.io/snaps/how-to/use-custom-ui/#spinner)

My question is how can we control it? Can we start it and after some stuff is done, just remove it? And maybe have buttons on dialog disabled when spinner is loading?

My use case is this:
1. Submit transaction to network
2. Show spinner - aka waiting for transaction to be confirmed on ledger
3. Show success / error message

**Comment by Montoya** [comment, marked as answer]
Hi! Great question. We haven't publicized the spinner component because there isn't a good way to do what you have in mind yet. We will eventually ship "dynamic UI" and then we will provide an example of how to do this. :)
